app/(onboard)/onboardingFlow.tsx

```diff
@@ -0,0 +1,187 @@
+import { View, Text, TouchableOpacity, Platform, Image, GestureResponderEvent } from 'react-native';
+import { useRouter, useLocalSearchParams} from 'expo-router';
+import React, { useState } from 'react';
+import FormInput from '@/components/FormInput';
+import AuthLayout from '@/components/AuthLayout';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import * as ImagePicker from 'expo-image-picker';
+
+import { useAuth, useUser } from '@clerk/clerk-expo';
+
+
+const onboardingFlow = () => {
+  const { email, sessionId } = useLocalSearchParams();
+  const router = useRouter();
+  const [step, setStep] = useState(0);
+  const [name, setFullName] = useState('');
+  const [photo, setPhoto] = useState<string | null>(null);
+  const [birthdate, setBirthdate] = useState(new Date());
```

Comment:
```suggestion
  const [userData, setUserData] = useState({
    name: '',
    birthdate: new Date(),
    photo: null,
  });
```

then a function like
```
const updateUserData = (key: string, value: any) => {
  setUserData((prev) => ({ ...prev, [key]: value }));
};
```

and use like   
```
onChangeText={(text: string) => updateUserData('name', text)}
```

or use `useReducer`